### DISCUSSION TITLE
使用ipv6竟然很多网站解析不了

### DISCUSSION THREAD

**Discussion by wpyok500** [SeaHOH/GotoX#55] [问答/咨询]
![12](https://user-images.githubusercontent.com/14154547/32150868-b1c0c628-bd52-11e7-8e1f-6070a841d202.png)
![13](https://user-images.githubusercontent.com/14154547/32150869-b34a892a-bd52-11e7-9d35-49ebd17abc7e.png)

**Comment by SeaHOH** [comment]
可参考 https://developers.google.com/speed/public-dns/docs/dns64 设置 DNS

或者，profile = ipv46

**Comment by SeaHOH** [comment]
修复了部分错误，实际结果没有变化，因为 DNS-over-HTTPS 解析是正确的。

**Comment by wpyok500** [comment]
ipv46解析规则很慢，个别网站一直解析不了。如：http://www.setn.com/  ；不管ipv6 DNS用的是google的还是OPENDNS的。
![image](https://user-images.githubusercontent.com/14154547/32156978-89613784-bd7b-11e7-8f65-bef2849227b7.png)

**Comment by SeaHOH** [comment]
`http://www.setn.com/` 只有 IPv4 地址。
如果要使用 IPv6 （`profile = ipv6`）连接它，请清空其它 DNS 服务器，设置成 DNS64。

~以上操作的前提是你有原生 IPv6，~ 不然就只能 ~设置成 `profile = ipv46`，然后~ 走 GAE。

**Comment by k79e** [comment]
用v4的也遇到了 
![2020-02-20_202610](https://user-images.githubusercontent.com/6483602/74936715-a3643d00-5425-11ea-8341-08b71a0243e6.png)
我不知道为啥 昨天是一个其他的网站也是红字 浏览器页面是个GAE的错误页面 有蓝条的.
今天那个红字网站又能打开了.

昨天红字的网站 一个是v4 only 一个是2个都有.

**Comment by SeaHOH** [comment]
> 用v4的也遇到了

只要不是始终如此，可以尝试重置 GotoX DNS 缓存、修改 DNS 设置、刷新系统 DNS 缓存等操作。
GotoX 内部缓存时间也可以调小一点。

另，github 相关域名建议使用固定 IP 列表，地址一般不会发生变化。

**Comment by k79e** [comment]
是这样啊. 昨天我看的那个站就是直连的. 走的fwd然后也没规则.
更奇怪的是报错页面咋是GAE的 真是有点怪了

刚才又遇到了个 不是github的 我重置了下软件的dns缓存 果然正常了
我发现访问那个网站前面他会报错一下 是黄字
![image](https://user-images.githubusercontent.com/6483602/75006537-0947d780-54ad-11ea-80e8-a93bb9a0f00c.png)

t.cn也是那样
然后对比正常访问时间发现多了2秒

抓本地回环发现1秒内 2个dns记录都返回了. 就是我设置的localdns和另一个dns.
软件同时发起了2个请求!!? 不知道为啥.(他向2个不同的dns发了请求 然后2个结果都收到了)
我的localhost = 0设置的是0 

localhost = 1的时候 t.cn的黄字消失了 这时候发起的dns请求是单个的(localservers)

可能localhost = 0的时候 两个dns写的不一样 就有bug了?
//然后localhost=0 localserver直接注释掉 访问t.cn还是第一次会有黄字. 奇怪了.
//这时候和不注释还是一样 他向2个dns发起了请求. 真是见鬼了!!!!!
//我的dns是127.0.0.1 这个是问题么? 我看到blacklist里面也有这个ip

更严重的
那个设置0 然后localserver删了 空白 他还是发起2个dns请求
一个127.0.0.1 一个127.0.0.2(servers)
.1的同时也是系统dns(备用 .2是第一个)
//我看到上面写的不要和系统设置重复... 我用了重复的 是这样会出问题么.
priority = system|remote|overhttps
如果就算重复了 那也是和system一样啊 难道他拿到system dns 的备用也会查询?

**Comment by k79e** [comment]
remote|system
还是发送了2个dns请求

remote
还是一样
这时候就算localserver 空 也还是发起2个dns请求. 都没填写他为什么要向127.0.0.1发请求呢!!!
//但是没黄字了

system
直接报错 黄字
还是和当初描述的一样 向2个dns发起了请求.

这区别就是既然都一样为啥有个报错了?

---------------------
优先级 remote
localhost = 1 localserver = 
这样的话无论cn还是其他域名 竟然都查不到dns流量了!!!!
//测了半天是有的域名有dns流量有的却没有 t.cn就是一直都没有.
//我就搞不清楚为啥这时候抓不到流量了 我开启软件 然后重置软件的缓存 还是一样 抓不到dns流量.

总之我发现localhost = 0的时候 根本这个设置是失效的.
得配置成=1 然后localserver 不填写才行

**Comment by SeaHOH** [comment]
估计是你 127.0.0.1 的那个 DNS 导致问题，你直接用其它工具看看它返回的结果，如 nslookup，dig。

- servers 用于解析域名，~会读取系统 DNS 设置~
- localservers 用于解析直连域名
- localhost 开启后，直连域名**只使用** localservers 解析
- localwhitelist 添加使用 localservers 解析的域名
- localblacklist 排除使用 localservers 解析的域名
- priority 会以默认顺序**补完**没有填写的优先级设置，同一优先级 server 的查询请求是**同时发出**的

**Comment by k79e** [comment]
真没问题.
localhost = 1的时候 不报错 0的时候报错.
都是发送了2个请求 都是请求收到了.....

blacklist我说的是那2个blacklist下面的blacklist
https://github.com/SeaHOH/GotoX/blob/master/config/Config.ini#L367

可是servers 配置文件里面说 不是给remote用的么. 那他为啥读取系统设置?
https://github.com/SeaHOH/GotoX/blob/master/config/Config.ini#L305
305行的设置还必须开启 直连的dns写空白 这时候就只发送一个了.

我感觉这主要是个奇怪的设置出错的问题 关键就在305那个地方. 
你说servers会读取系统设置 那为啥他又可以用手写.

**Comment by SeaHOH** [comment]
> blacklist我说的是那2个blacklist下面的blacklist

和它没有关系。

> 你说servers会读取系统设置 那为啥他又可以用手写.

手滑，刚才写错了。

**Comment by k79e** [comment]
我发现报错的必须用system优先级 
只用这个或者他第一才会这样.
用其他的就没问题.
出问题的时候 localhost=0 才会报错. =1不会. 哪怕localserver是空白也没事.

**Comment by SeaHOH** [comment]
你的描述不太清晰，看得有点晕。你能把具体 DNS IP 提供下吗？哪个解析会出错，哪个不会。如果是本机服务，请提供软件和上游。

**Comment by k79e** [comment]
servers = 127.0.0.2:53
localservers = 127.0.0.1:53
localhost = 0
priority = system
就这3条
//反正localhost配置的都是0 所以第二行应该根本不生效才对.
第一行我估计填什么都一样

**Comment by k79e** [comment]
我更新了描述.
重新测试了 第一个填啥都一样.... 我填了tx的dns还是一样. 第一次请求t.cn 直接黄字....
然后清除dns缓存 不重启软件 继续请求 还是冒黄字. 然后他会自动重试 然后能正常范围.
不过黄字一出现 他会卡个一两秒......
还好今天我研究出来了避免黄字的办法!!!

**Comment by k79e** [comment]
servers = 119.29.29.29
//localservers = 127.0.0.1:53
localhost = 0
priority = system
你可以用这个测试

我又开了流量截取 发现黄字出现一瞬间 根本没发出dns请求.
dns请求出现的时候 内容已经载入了 说明出现dns流量的时候 是第二次重试的结果.

第一次是瞬间报错 估计软件什么地方有问题...

报错信息是这样的
![image](https://user-images.githubusercontent.com/6483602/75040218-3cae5480-54f5-11ea-9da2-3037de16a121.png)
其实我感觉这连接被重置根本就是文不对题吧... 因为根本就不存在连接.....
如果真是连接方面的问题 我应该抓到发送2个dns请求才对 第一个有问题 第二个没有才是.
然而不是那样.

**Comment by k79e** [comment]
127.0.0.2
127.0.0.1
.............
这就更不对劲了. 我用了system 他凭什么查询119.29.29.29的!!?
我2个系统的都作废了不成?
不可能的 dig查询无论哪个地址都没问题!!!! 我天天都用的哇. 如果有问题我早没法上网了.
我看下回环流量去.... 等会更新下...

2020-02-21 22:08:12
本地回环流量是空白的 没有请求我本地的任何dns 所以这问题你问了也白问把?
出问题的话应该看到请求无回答才是

**Comment by k79e** [comment]
更新描述. 
.2的是v2的分流dns 然后国内的会跳会.1的
.1的是阿克力缓存 :D 好像叫acrylic
上游还是tx dns 119的和1.2.4.8的
由于阿克力缓存是并向请求 所以如果是他的话 根本不用看本地回环流量.
会看到2个dns一起发到公网上去.... 然而并没有情况....

https://github.com/SeaHOH/GotoX/issues/55#issuecomment-589662756
所以这个设置他根本没有使用系统dns 反而还用了remote配置.
应该是设置开关有点错误.

**Comment by k79e** [comment]
你自己看么 抓到了本地的那个http
![image](https://user-images.githubusercontent.com/6483602/75041270-36b97300-54f7-11ea-8fea-960676ae6570.png)

![image](https://user-images.githubusercontent.com/6483602/75041259-30c39200-54f7-11ea-8998-4c588cd66ebc.png)
 dns白的

所以2个设置开关有诡异的问题.

**Comment by SeaHOH** [comment]
我到觉得你的配置不太合适。
试试
```ini
servers = 127.0.0.2
localservers = 127.0.0.1
localhost = 1
priority = remote|overhttps|system
```

**Comment by k79e** [comment]
https://github.com/SeaHOH/GotoX/issues/55#issuecomment-589662756
还是那个地方 改成localhost=1
也就是
servers = 119.29.29.29
localservers = 127.0.0.1:53
#是否使用 localservers 解析直连域名
localhost = 1
优先级用的remote only

~//这次换了公网的 然后查询发现没有公网流量~
我干脆把localserver改成114的那个了
~2个都是公网 然而请求之后依然没有公网流量~
总之remote在前面是没什么问题的 就是改成本地ip会变成并行请求



//remote优先级在前 开localhost是没问题的
 //localhost关 发现就只用119的了.
~//同时如果把ip换成本地的 会发现流量是并行的.~
servers = 127.0.0.2
localservers = 127.0.0.1
localhost = 0
priority = remote
![image](https://user-images.githubusercontent.com/6483602/75042410-4fc32380-54f9-11ea-9005-7b76b1326cc0.png)

不对 真是见鬼了
是本地缓存没有缓存跳过去的请求. .1的请求是.2的程序发的. 晕啊...

**Comment by SeaHOH** [comment]
> 优先级用的remote only

没有 only，会补全。



> 总之remote在前面是没什么问题的 就是改成本地ip会变成并行请求

不是哪个选项在前的问题，而是实际查询顺序是怎样的，因为 IP 是由用户顺便写的，没人能限制。

**Comment by k79e** [comment]
真有问题啊 system在前面的确每次都出问题.
而且他根本不用system的ip呢

**Comment by k79e** [comment]
servers = 119.29.29.29
localservers = 114.114.114.114
localhost = 0
priority = system|remote|overhttps
就这个设置 他就重试时候用的119的 第一次直接就瞬间报错 根本不发请求就报错了.
就是我给你发的那个dns白板 根本没有dns流量 这你咋解释?
既然并行的也不是这个软件发送的. 可是另一个情况 他都不发流量就报错这是为啥

**Comment by SeaHOH** [comment]
你的系统 DNS 里面是 127.0.0.1-2，而且有递归关系，不要把 system 设置到前面。
你按这个 https://github.com/SeaHOH/GotoX/issues/55#issuecomment-589671464 设置绝对没有问题。

**Comment by SeaHOH** [comment]
这些设置的目的就两个，分流和后备。不要分流错误和重复请求。

**Comment by k79e** [comment]
请问这个localhost设置是属于remote的还是system的?
我真是搞不清.

**Comment by SeaHOH** [comment]
localhost 开启后，直连域名只使用 localservers 解析，反之，localservers 完全无用。

**Comment by k79e** [comment]
我建议红字的解析失败 不缓存他. 要不然有时候失败了还得每次手动重置dns

**Comment by k79e** [comment]
> 反之，localservers 完全无用。

那这就是最奇怪的问题 那个黄字就要求关闭localhost 然后system在前最面

**Comment by SeaHOH** [comment]
> 我建议红字的解析失败 不缓存他. 要不然有时候失败了还得每次手动重置dns

这个必须缓存，时间现在是 300 秒，可以稍微减小些。

> 那这就是最奇怪的问题 那个黄字就要求关闭localhost 然后system在前最面

配合 localservers/localhost 使用，不影响其它设置
localwhitelist 添加使用 localservers 解析的域名
localblacklist 排除使用 localservers 解析的域名

你可以使用这个功能。

**Comment by k79e** [comment]
网站不存在 是红字
网络卡了 他超时了也是红字....
